File: hadoop/src/ContadorDePalavrasMapper.java
import org.apache.hadoop.io.IntWritable;
import org.apache.hadoop.io.LongWritable;
import org.apache.hadoop.io.Text;
import org.apache.hadoop.mapreduce.Mapper;

import java.io.IOException;

public class ContadorDePalavrasMapper extends Mapper<LongWritable, Text, Text, IntWritable> {
    private final static IntWritable one = new IntWritable(1);
    private Text word = new Text();

    @Override
    protected void map(LongWritable key, Text value, Context context) throws IOException, InterruptedException {
        // Divide a linha de texto em palavras
        String[] words = value.toString().split("\\s+");

        // Emite cada palavra com o valor 1
        for (String str : words) {
            word.set(str);
            context.write(word, one);
        }
    }
}

File: hadoop/src/ContadorDePalavrasMapper.scala
import org.apache.hadoop.io.{IntWritable, LongWritable, Text}
import org.apache.hadoop.mapreduce.Mapper

class ContadorDePalavrasMapper extends Mapper[LongWritable, Text, Text, IntWritable] {
  val one = new IntWritable(1)
  val word = new Text()

  override def map(key: LongWritable, value: Text, context: Mapper[LongWritable, Text, Text, IntWritable]#Context): Unit = {
    val line = value.toString
    val tokens = line.split("\\s+")
    tokens.foreach { token =>
      word.set(token)
      context.write(word, one)
    }
  }
}
